apps/music/test/marionette/Player_test.js

```diff
@@ -267,6 +267,37 @@ marionette('Music player tests', function() {
     });
   });
 
+  suite('Repeat test', function() {
+    test('Check that repeat button works', function() {
+      try {
+        music.launch();
+        music.waitForFirstTile();
+        music.switchToSongsView();
+        music.playFirstSong();
+
+
+        assert.equal(music.getRepeatSetting(), 'off');
+        music.tapRepeat();
+        assert.equal(music.getRepeatSetting(), 'list');
+        music.tapRepeat();
+        assert.equal(music.getRepeatSetting(), 'song');
+
+        // close the app because we want to test things are saved.
+        music.close();
+
+        // start it over.
+        music.launch();
+        music.waitForFirstTile();
+        music.switchToSongsView();
+        music.playFirstSong();
+
+        assert.equal(music.getRepeatSetting(), 'song');
+      } catch(e) {
+        assert.ok(false, 'Exception: ' + e.stack);
```

Comment:
you no longer need these try { } catch() blocks since the test infrastructure has now be fixed and the stack traces are properly reported.